### DISCUSSION TITLE
OBAL on Raspberry Pi Zero W 2 and Jetson Nano

### DISCUSSION THREAD

**Discussion by b06stiles** [HefnySco/OBAL#24] [Q&A] [answered]
Hi Mohammad

Firstly thanks for your great work!  I now have some OBAL boards waiting to be built!

Question: Has anyone successfully got Ardupilot running on the Pi Zero W 2?

I've had success on the Zero W (Debian Bullseye), and partial success on both the Zero W 2 and the Jetson Nano using the aarch64-linux-gnu compiler, but the results and problems may be out of scope of this forum...

Many thanks

Bob

**Comment by FelixvdDonk** [comment]
https://www.youtube.com/watch?v=7T92mprZYnQ In this video it is running on the Pi Zero W 2.

**Reply by b06stiles** [reply]
Thanks for that - I didn't look closely enough!

**Comment by HefnySco** [comment, marked as answer]
@b06stiles OBAL - especially [GY-91 version ](https://github.com/HefnySco/OBAL/wiki/07---OBAL-GY91) runs best on RPI-Z2.
You can even use RPI-camera while running ardupilot, but in this case you need to use [cpu affinity](https://github.com/HefnySco/OBAL/wiki/06---Build-and-Install#cpu-affinity) to dedicate a cpu to ardupilot.

I believe the problems you faced is because using 64-bit OS or Bullseye.  I use the legacy [Raspberry Pi OS (Legacy)
](https://downloads.raspberrypi.org/raspios_oldstable_lite_armhf/images/raspios_oldstable_lite_armhf-2022-09-26/2022-09-22-raspios-buster-armhf-lite.img.xz).
For Jetson nano I have no idea. Please share your experience with Jetson Nano when possible.

**Reply by b06stiles** [reply]
Hm sounds right.  I was hoping that the 64 bit OS would be backward compatible.

For the record, the part that is failing is the call to mmap() in the Memory_Table constructor which maps physical addresses to virtual memory (in RCInput_RPI.cpp).  It's returning 0xFF....FF, which is then crashing memset(). 

I'll try 32-bit Bullseye, and if that fails, the link above.
Thanks!

**Reply by HefnySco** [reply]
Thank you for sharing this info.

**Reply by b06stiles** [reply]
OK Ardupilot is now running on the Pi Zero W 2 using both Buster 32 bit and Bullseye 32 bit lite.  Cool!
Thanks guys.
